wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -340,52 +340,109 @@ public function getFieldHtml()
+                }
             } else {
-                $value = $data[$this->getObjectProperty()];
-                $method = "variable '" . \get_class($object) . "::$" . $this->getObjectProperty() . "'";
+                $variableName = $this->getObjectProperty();
+                try {
+                    $value = $this->readUploadLocations($data[$variableName] ?? null);
+                } catch (\InvalidArgumentException | \TypeError $e) {
+                    throw new \InvalidArgumentException("The property {$className}::\${$variableName} must contain an array of strings with the valid file locations for field '{$this->getId()}'.", 0, $e);
+                }
             }
 
-            if (\is_array($value)) {
-                $value = \array_map(function ($v) use ($method) {
-                    if (!\is_string($v) || !\file_exists($v)) {
-                        throw new \InvalidArgumentException(
-                            "The {$method} must return an array of strings with the file locations for field '{$this->getId()}'."
-                        );
-                    }
-
-                    return new UploadFile(
-                        $v,
-                        \basename($v),
-                        ImageUtil::isImage($v, \basename($v), $this->svgImageAllowed()),
-                        true,
-                        $this->svgImageAllowed()
-                    );
-                }, $value);
+            $this->value($value);
+        }
 
-                $this->value($value);
-            } else {
-                throw new \InvalidArgumentException(
-                    "The {$method} must return an array of strings with the file locationsfor field '{$this->getId()}'."
-                );
+        return $this;
+    }
+
+    /**
+     * Reads an array of `UploadFile` objects and validates it.
+     *
+     * @param UploadFile[] $data
+     * @return UploadFile[]
+     * @throws \InvalidArgumentException
+     * @since 5.5
+     */
+    private function readUploadFiles(array $data): array
+    {
+        foreach ($data as $value) {
+            if (!($value instanceof UploadFile)) {
+                throw new \InvalidArgumentException("At least one value is not an instance of '" . UploadFile::class . "'.");
             }
         }
 
-        return $this;
+        return $data;
+    }
+
+    /**
+     * Reads an array of strings containing file locations and validates it.
+     *
+     * @param string[] $data
+     * @return UploadFile[]
+     * @throws \InvalidArgumentException
+     * @since 5.5
+     */
+    private function readUploadLocations(array $data): array
+    {
+        foreach ($data as $key => $value) {
+            if (!\is_string($value)) {
+                throw new \InvalidArgumentException('At least one value is not a string.');
+            } elseif (!\file_exists($value)) {
+                throw new \InvalidArgumentException("File '{$value}' could not be found.");
+            }
```

Comment:
```suggestion
            }
            if (!\file_exists($value)) {
                throw new \InvalidArgumentException("File '{$value}' could not be found.");
            }
```
I'd avoid `else` if the `if` forces the function to return.